examples/external-dns/main.tf

```diff
@@ -0,0 +1,123 @@
+## locals
+
+locals {
+  plan_id = data.duplocloud_tenant.current.plan_id
+}
+
+## Datasources
+
+# Get tenant info, typically from the current workspace
+data "duplocloud_tenant" "current" {
+  name = var.tenant_name
+}
+
+# Get EKS credentials for helm provider
+data "duplocloud_eks_credentials" "current" {
+  plan_id = local.plan_id
+}
+
+# Get infrastructure to get the region for aws provider
+data "duplocloud_infrastructure" "current" {
+  infra_name = local.plan_id
+}
+
+# Get the route53 zone information
+data "aws_route53_zone" "selected" {
+  name = var.public_dns_domain
+}
+
+## Provider versions
+
+terraform {
+  required_version = ">= 1.4.4"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.12.0"
+    }
+    duplocloud = {
+      source  = "duplocloud/duplocloud"
+      version = ">= 0.10.0"
+    }
+    helm = {
+      source  = "hashicorp/helm"
+      version = ">= 2.11.0"
+    }
+  }
+  backend "s3" {
+    workspace_key_prefix = "duplocloud/addons"
+    key                  = "external-dns"
+    encrypt              = true
+  }
+}
+
+## Provider configs
+
+# Configure helm provider for chart deployment
+provider "helm" {
+  kubernetes {
+    host                   = data.duplocloud_eks_credentials.current.endpoint
+    cluster_ca_certificate = data.duplocloud_eks_credentials.current.ca_certificate_data
+    token                  = data.duplocloud_eks_credentials.current.token
+  }
+}
+
+# Configure aws provider for IAM role extension
+provider "aws" {
+  region = data.duplocloud_infrastructure.current.region
+}
+
+## Modules and resources
+
+# Deploy external-dns module for managing Route53 records.  
+# See https://github.com/kubernetes-sigs/external-dns/tree/master/docs/tutorials
+module "external_dns" {
+  source       = "../../modules/external-dns"
+  tenant_name  = var.tenant_name
+  values = {
+    "aws.zone" : "public",
+    "txtOwnerId" : data.aws_route53_zone.selected.id
+    "domainFilters[0]" : var.public_dns_domain
+  }
+}
+
+# Default duplo tenant role doesn't have Route53 permissions.  Add them here.
+data "aws_iam_policy_document" "service_extension" {
```

Comment:
This can be part of the module. I believe this could be a good thing to make dynamic. Like `aws.enabled` means use the aws values and make the policy and attach to the tenant role. Then we don't need to to trust users to build the role correctly.

Reply:
I didn't add this extension to the module because we can easily run afoul of policy attachment limits on the tenant role.  User's may be maintaining amalgamated IAM policies in their tenant terraform already.  I take your point that this is definitely a requirement for external-dns so if you think it's advisable to just maintain a separate policy I'm happy to add it.

Reply:
Spoke to kelly, will implement iam policy as we discussed